saw-script/src/SAWScript/Interpreter.hs

```diff
@@ -4551,6 +4551,10 @@ primitives = Map.fromList
     , "if the inner action does NOT raise an exception. This is primarily used"
     , "for unit testing purposes, to ensure that we can elicit expected"
     , "failing behaviors."
+    , ""
+    , "Note: to catch a failure that occurs in non-monadic code, you must wrap"
+    , "the non-monadic code in a do-block; otherwise it will be evaluated before"
+    , "fails runs."
```

Comment:
This is subtle enough that I wonder if we should add a manual entry about catching errors in code, making sure to mention at least the potential pitfall here. For now, such a manual entry would only describe `fails`, but I imagine that we might extend it later to talk about alternative mechanisms that catch errors in pure code (https://github.com/GaloisInc/saw-script/issues/2424).